examples/Components/Typeahead/Typeahead.purs

```diff
@@ -2,178 +2,198 @@ module Docs.Components.Typeahead where
 
 import Prelude
 
-import Effect.Aff.Class (class MonadAff)
 import Data.Array (elemIndex, mapWithIndex, difference, filter, (:))
-import Data.Foldable (length, traverse_)
+import Data.Const (Const)
+import Data.Foldable (for_, length)
 import Data.Maybe (Maybe(..))
+import Data.Monoid (guard)
 import Data.String (Pattern(..), contains)
+import Data.Symbol (SProxy(..))
+import Data.Variant (Variant, inj, match)
+import Docs.CSS as CSS
+import Effect.Aff.Class (class MonadAff)
+import Effect.Class.Console as Console
 import Halogen as H
 import Halogen.HTML as HH
 import Halogen.HTML.Events as HE
 import Halogen.HTML.Properties as HP
 import Select as Select
 import Select.Setters as Setters
 
-import Docs.CSS as CSS
-
 type TypeaheadItem = String
 
-data Query a
-  = Log String a
-  | HandleInputContainer (Select.Message Query TypeaheadItem) a
-  | Removed TypeaheadItem a
+data Action
+  = Remove TypeaheadItem
+  | HandleSelect (Select.Message TypeaheadItem ())
```

Comment:
For example, this component no longer has to handle the new action (`Log`) as part of its eval, no `Emit`, nothing. Instead, it provides its own handler to run inside the renderless component.